arbitrator/prover/src/memory.rs

```diff
@@ -71,7 +72,7 @@ pub struct Memory {
     pub merkle: Option<Merkle>,
     pub max_size: u64,
     #[serde(skip)]
-    dirty_indices: Mutex<HashSet<usize>>,
+    dirty_leaves: Mutex<HashSet<usize>>,
```

Comment:
This can be done in a follow-up PR, but it might be worth performance testing a bitvec here as well instead of a hashset

Reply:
I added this task to a comment on the linear ticket: https://linear.app/offchain-labs/issue/NIT-2411/arbitrator-optimizations-to-reduce-number-of-challenge-levels-to-3#comment-86e91b86